.github/workflows/deploy.yml

```
@@ -0,0 +1,126 @@
+    - name: Checkout code
+      uses: actions/checkout@v3
+    
+    - name: Log in to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v4
```

actionlint (1.7.7)

the runner of "docker/build-push-action@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)